misc/python/materialize/output_consistency/input_data/params/number_operation_param.py

```diff
@@ -37,3 +48,29 @@ def __init__(
             incompatibilities,
             incompatibility_combinations,
         )
+        self.only_int_type = only_int_type
+        self.no_int_type_larger_int4 = no_int_type_larger_int4
+        self.no_floating_point_type = no_floating_point_type
+        self.no_unsigned_type = no_unsigned_type
+
+    def supports_type(self, data_type: DataType) -> bool:
+        if not isinstance(data_type, NumberDataType):
+            return False
+
+        if self.only_int_type and data_type.is_decimal:
+            return False
+
+        if self.no_int_type_larger_int4 and data_type.identifier in {
+            INT8_TYPE_IDENTIFIER,
+            UINT4_TYPE_IDENTIFIER,
```

Comment:
Is uint4 larger than int4? Is this about the MAX value that fits inside?

Reply:
Yes, it is about the max value. Any alternative naming suggestions?

Reply:
Ok, my interpretation was that it uses more bytes, so uint4 and int4 would be the same size.
I don't have a good term, `type_fits_larger_values_than_INT4_MAX` seems a bit unwieldy.